src/osp/Active/SysVehicle.cpp

```diff
@@ -276,136 +192,144 @@ float SysVehicle::compute_hier_volume(ActiveScene& rScene, ActiveEnt part)
 
-    /* A list of MachineDefs, which catalog all part machines and the ActiveEnts
-     * that are created to represent the objects that own them
-     */
-    std::vector<MachineDef> machineMapping;
-    machineMapping.reserve(prototypes.size());
 
-    SPDLOG_LOGGER_TRACE(m_scene.get_application().get_logger(), "New entities size", newEntities.size());
+    // reserve space for new entities and ACompTransforms to be created
+    rScene.get_registry().reserve(
+                rScene.get_registry().capacity() + part.m_entityCount);
+    rScene.get_registry().reserve<ACompTransform>(
+                rScene.get_registry().capacity<ACompTransform>() + part.m_entityCount);
 
-    for (size_t i = 0; i < prototypes.size(); i++)
+    // Create entities and hierarchy
+    for (size_t i = 0; i < part.m_entityCount; i++)
     {
-        PrototypeObject const& currentPrototype = prototypes[i];
+        PCompHierarchy const &pcompHier = part.m_partHier[i];
         ActiveEnt parentEnt = entt::null;
 
         // Get parent
-        if (currentPrototype.m_parentIndex == i)
+        if (pcompHier.m_parent == i)
         {
-            // if parented to self,
+            // if parented to self (the root)
             parentEnt = rootParent;//m_scene->hier_get_root();
         }
         else
         {
             // since objects were loaded recursively, the parents always load
             // before their children
-            parentEnt = newEntities[currentPrototype.m_parentIndex];
+            parentEnt = newEntities[pcompHier.m_parent];
         }
 
         // Create the new entity
-        ActiveEnt currentEnt = rScene.hier_create_child(parentEnt,
-                                                currentPrototype.m_name);
+        ActiveEnt currentEnt = rScene.hier_create_child(parentEnt);
         newEntities[i] = currentEnt;
 
         // Add and set transform component
-        ACompTransform& currentTransform
-                = rScene.reg_emplace<ACompTransform>(currentEnt);
-        currentTransform.m_transform
-                = Matrix4::from(currentPrototype.m_rotation.toMatrix(),
-                                currentPrototype.m_translation)
-                * Matrix4::scaling(currentPrototype.m_scale);
-
-        if (currentPrototype.m_type == ObjectType::MESH)
-        {
-            using Magnum::GL::Mesh;
-            using Magnum::Trade::MeshData;
-            using Magnum::GL::Texture2D;
-            using Magnum::Trade::ImageData2D;
+        PCompTransform const &pcompTransform = part.m_partTransform[i];
+        auto &rCurrentTransform = rScene.reg_emplace<ACompTransform>(currentEnt);
+        rCurrentTransform.m_transform
+                = Matrix4::from(pcompTransform.m_rotation.toMatrix(),
+                                pcompTransform.m_translation)
+                * Matrix4::scaling(pcompTransform.m_scale);
+    }
+
+    // TODO: add an ACompName because putting name in hierarchy is pretty stupid
```

Comment:
`ACompHierarchy::m_name` is only used in a handful of places so far, it would be reasonable to include this change in this PR while you're at it if you like.